src/software/estop/threaded_estop_reader.cpp

```diff
@@ -0,0 +1,89 @@
+
+#include "threaded_estop_reader.h"
+
+#include <boost/bind/bind.hpp>
+#include <boost/function.hpp>
+#include <cstdio>
+#include <cstring>
+#include <thread>
+#include <utility>
+
+#include "software/logger/logger.h"
+
+
+ThreadedEstopReader::ThreadedEstopReader(int startup_ms, int interval_ms,
+                                         std::unique_ptr<UartCommunication> uart_reader)
+    : startup_ms(startup_ms),
+      interval_ms(interval_ms),
+      timer(io_service, this->interval_ms),
+      estop_msg(0),
+      uart_reader(std::move(uart_reader)),
+      estop_state(EstopState::STOP)
+{
+    estop_thread = std::thread(boost::bind(&ThreadedEstopReader::continousRead, this));
+}
+
+void ThreadedEstopReader::continousRead()
+{
+    timer.async_wait(
+        boost::bind(&ThreadedEstopReader::tick, this, boost::asio::placeholders::error));
+    io_service.run();
+}
+
+EstopState ThreadedEstopReader::getEstopState()
+{
+    return estop_state;
```

Comment:
this isn't thread safe since `continousRead` modifies the `estop_state` in another thread, and `getEstopState` doesn't grab any locks

maybe use an `atomic_bool` here as well? the `STATUS_ERROR` state doesn't seem to have a recovery action that can be taken by the caller of `getEstopState()` so maybe the enum can just be replaced with an atomic bool

Reply:
method now changed to `isEstopPlay()` which returns bool. it is now thread safe as `estop_state` is grabbed with a lock